### ISSUE TITLE
Increase threshold or make it configurable in the user interface

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/image_resize_filter#7]
I'm currently testing this nice module, and one of the first things I saw was the 'threshold' message:

> Image resize threshold of 5 per minute reached. Some images have not been resized. Resave the content to resize remaining images.

The threshold provides a "[basic flood prevention](https://github.com/backdrop-contrib/image_resize_filter/blob/c57f7460d7f587d659986face2807aaef80abcba/image_resize_filter.module#L345)", but five images can easily be reached when you edit a post with some inline images. I'd suggest to increase the limit and / or make it configurable in the user interface.

**Comment by olafgrabienski** [comment]
Side note / question: As a workaround, can I increase the limit directly in the active configuration file `image_resize_filter.settings.json`, or is this not recommended?

**Comment by indigoxela** [comment]
I don't think it does any harm. It's just a number in a json file. :wink:

A UI for that setting would be a nice addition.

**Comment by laryn** [comment]
I'm not opposed to adding this. I may be doing a little work on this module later this month and if so I'll try to include this.

**Comment by laryn** [comment]
@olafgrabienski Are you able to test the linked PR? (Or @indigoxela)

**Comment by olafgrabienski** [comment]
@laryn Thanks for providing the PR! It seems to work fine: Setting the maximum (from the default `10`) to low numbers like `1` or `2` leads to expected error messages that some images have not been resized. Setting the number back makes the resize for more images working again.

There is one small issue, though: When I set the maximum number to `1`, the message is about a threshold of `0` per minute. A max of `2` leads to a treshold of `1`, and so forth.

I have also a question: The label in the settings dialog is "Maximum number of remote images to resize". What's the meaning of "remote" here?

**Comment by laryn** [comment]
@olafgrabienski I made a few tweaks for messaging -- does that feel better? Or if you want to figure out a better way to phrase things I'm open! 

For background, it looks like the threshold is per two-minutes, and the message is dividing and rounding to get a per minute number.

This is the closest thing to an issue in the Drupal module:

 - https://www.drupal.org/project/image_resize_filter/issues/2196623

This flood protection code was added as part of this potential DDOS issue:

 - https://www.drupal.org/node/2194655

**Comment by olafgrabienski** [comment]
Hey @laryn, the changed / added texts look good to me. Thanks again, also for the related Drupal information.

**Comment by laryn** [comment]
Thanks @olafgrabienski -- I've merged this.

I should also note that as part of the work on this form I cleaned up a few adjacent things (stripped out a theme implementation, added a small CSS file on the form page to style things without the theme implementation, and removed a javascript file that could easily be replaced using `#states`). In retrospect I should perhaps have filed those as separate issues.
